### PR TITLE
New GUI

### PR DESCRIPTION
**Pull request by OpenSourceSimon**
# Description
It's almost here, a full GUI. Edit, see, and start the bot easily with the new GUI. Not 100% done, we're almost there. Made by @VirtualRabb and me

# Screenshots
image
image

# Checklist:

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

# Any other information (e.g how to test the changes)
Feel free to test the code we have so far, and give feedback on how we can make it better.

# Todo
- [x] Import all the settings on the startup
- [x] Save the settings that changed to config.toml
- [ ] View a list of the videos in the results tab
- [ ] Progression bar

### CODE REVIEW

**Comment by JasonLovesDoggo** [comment]
Don't forget to implement a good way for queuing post's via the post+post+post format

**Comment by StopmotionSimonYT** [comment]
> Don't forget to implement a good way for queuing post's via the post+post+post format

Thanks for the feedback! Noted!

File: GUI.py [hunk 1]
```diff
@@ -1,128 +1,905 @@
...
+
+        if type == "googletranslate":
+            self.tts_google_translate_voice_title.tkraise()
+            self.tts_google_translate_voice.tkraise()
+
+        if type == "awspolly":
+            self.tts_aws_polly_voice_title.tkraise()
+            self.tts_aws_polly_voice.tkraise()
+
+        if type == "pyttsx":
```
Comment (JasonLovesDoggo) [inline GUI.py:862]: Try not to hardcode this to each option

**Review by JasonLovesDoggo — CHANGES_REQUESTED** [review]

File: GUI.py (hunk 1 above)
Comment (StopmotionSimonYT) [reply GUI.py:882]: Hi,
Thanks for reviewing our code! I understand it. We're working on it, but the tkraise can't be a variable. So it's difficult to do it. Do you have a suggestion for us on how we can do it?

**Comment by CJthisis** [comment]
Im just curious: What is missing from the GUI to be added to the develop branch?

**Comment by OpenSourceSimon** [comment]
> Im just curious: What is missing from the GUI to be added to the develop branch?

There isn't a progress and results page yet. We can merge it without that, but then it needs to be tested first.

**Comment by fin-github** [comment]
You are crazy. I'm pretty popular on tiktok but now that I see this I could improve my productivity so much. This is awesome!!!

**Comment by OpenSourceSimon** [comment]
> You are crazy. I'm pretty popular on tiktok but now that I see this I could improve my productivity so much. This is awesome!!!

Glad I could help! Feel free to test it and let me know if you need help!

**Comment by fin-github** [comment]
Can you help me download it, I almost never use pull requests lmao.

**Comment by fin-github** [comment]
> > You are crazy. I'm pretty popular on tiktok but now that I see this I could improve my productivity so much. This is awesome!!!
> 
> Glad I could help! Feel free to test it and let me know if you need help!

can you help me get/test this pull request? I dont know how

**Comment by fin-github** [comment]
How do you test this code?

**Comment by CJthisis** [comment]
> How do you test this code?

You just download the branch, in this case
https://github.com/OpenSourceSimon/RedditVideoMakerBot/tree/GUI

(Assuming you're on Desktop) you simply click on the green "Code" icon and click on "Download zip". Then you can start the bot like you normally do, but keep in mind that some things that are in the develop branch are not yet implemented in this version.

**Comment by fin-github** [comment]
> 

Hey! Thanks a lot for replying. I got it installed and working but with one problem. I cant use the gui. I ran the `py GUI.py` command and it opens the localhost website, but with one problem. It shows this:
image
. Can you help me resolve this? Did I mess something up when installing?
Heres the output in a photo:
image
. If you can see that I will put the output in text: `C:\Users\finle\OneDrive\Documents\RedditVideoMakerBot-master>gui.py
Server started at http://localhost:4000/GUI/
Website opened in new tab
Press Ctrl+C to quit
127.0.0.1 - - [29/Dec/2022 22:42:25] "GET /GUI/ HTTP/1.1" 304 -
127.0.0.1 - - [29/Dec/2022 22:42:25] "GET /video_creation/data/videos.json HTTP/1.1" 304 -
127.0.0.1 - - [29/Dec/2022 22:42:33] "GET / HTTP/1.1" 200 -`

**Please help!**

**Comment by CJthisis** [comment]
Ok, this is actually quite a simple problem to solve. It looks like you haven't run the bot before, and All you need to do is run the bot through main.py, this is the way the bot currently works.

It's going to ask you for a bunch of details, which will then be saved in a new file called toml.config, which you can open with a text editor to quickly change settings.

Hope this helps!

**Comment by OpenSourceSimon** [comment]
> Ok, this is actually quite a simple problem to solve. It looks like you haven't run the bot before, and All you need to do is run the bot through main.py, this is the way the bot currently works.
> 
> It's going to ask you for a bunch of details, which will then be saved in a new file called toml.config, which you can open with a text editor to quickly change settings.
> 
> Hope this helps!

Yes, thanks for your reply! The folder name he's using is from the master branch, which can be the problem. @fin-github I recommend joining the Lewis discord server so we can assist you better!

**Comment by fin-github** [comment]
> > Ok, this is actually quite a simple problem to solve. It looks like you haven't run the bot before, and All you need to do is run the bot through main.py, this is the way the bot currently works.
> > It's going to ask you for a bunch of details, which will then be saved in a new file called toml.config, which you can open with a text editor to quickly change settings.
> > Hope this helps!
> 
> Yes, thanks for your reply! The folder name he's using is from the master branch, which can be the problem. @fin-github I recommend joining the Lewis discord server so we can assist you better!

Ok I will be joining!

**Comment by CJthisis** [comment]
Is the GUI not planned for the future anymore?

**Comment by OpenSourceSimon** [comment]
> Is the GUI not planned for the future anymore?

We want to merge it, but it's not finished. I don't have time to finish it and the co-maker doesn't reply anymore. Feel free to finish it if you want

**Comment by mallik30** [comment]
this looks cool, please add a voice play button next to voice to listen to different voices before selecting

**Comment by fin-github** [comment]
Hey. I realized you stopped this. I will make this a project of mine, I will start making it again.

**Comment by fin-github** [comment]
Right now I am working on the results page that you marked unmade in the TODO

**Comment by OpenSourceSimon** [comment]
> Right now I am working on the results page that you marked unmade in the TODO
> 

Awesome! Let me know if you have any questions, and feel free to join Lewis' Discord server if you haven't already.

**Comment by joshholly** [comment]
Please update the new GUI to work on the new 3.2 config files . We need to be able to add background audio. Also background_choice is now background_video / background_audio

**Pull request closed without merging**